### ISSUE TITLE
Approximate Unique Count

### ISSUE THREAD

**Issue by nikolaydubina** [gjbae1212/hit-counter#44]
I would really interested in unique users. Even approximately.

How about using IP + [HyperLogLog](https://redis.io/docs/data-types/probabilistic/hyperloglogs/)?

That already would be massively better!

@gjbae1212 what you think?

**Comment by nikolaydubina** [comment]
cc @gjbae1212

**Comment by achrafaitibba** [comment]
Or at least using sessionId, because counts should also count users that visits same content from time to time

**Comment by nikolaydubina** [comment]
how would you pass session id? through cookie or something? how that would work?

if you need javascript, that is too much. as hitcounter is best since it works in environments without javascript (e.g. emails, image html embeddings in markdown, etc.)

**Comment by achrafaitibba** [comment]
Yeah, makes sense. If JS is off the table, then cookies won’t work. Maybe using an IP+User-Agent hash? Not perfect but better than raw hits. Keeps it lightweight without javascript.
